File: javascript/calculator_cli_app/__tests__/queries/calculation_query_test.js
"use strict;"

import CalculationQuery from "../../src/queries/calculation_query";

const calculation_query = new CalculationQuery("foo");

test("test_f_with_zero", () => {
  expect(calculation_query.f(0)).toBe(1);
});

test("test_f_with_two", () => {
  expect(calculation_query.f(2)).toBe(2);
});

test.skip("test_f_with_four", () => {
  expect(calculation_query.f(4)).toBe(348);
});

File: ruby/calculator_cli_app/test/queries/calculation_query_test.rb
require 'minitest/autorun'
require_relative '../../src/queries/calculation_query'

class CalculationQueryTest < Minitest::Test
  def setup
    @calculation_query = ::Queries::CalculationQuery.new('foo')
  end

  private

  attr_reader :calculation_query
end

class ArgumentZeroTest < CalculationQueryTest
  def test_f
    assert_equal(1, calculation_query.f(0))
  end
end

class ArgumentTwoTest < CalculationQueryTest
  def test_f
    assert_equal(2, calculation_query.f(2))
  end
end

class ArgumentFourTest < CalculationQueryTest
  def test_f
    assert_equal(348, calculation_query.f(4))
  end
end
